### PR TITLE
Document zcml:condition more thoroughly.

### LINKED ISSUE #5: The zcml namespace is not in the apidoc
In https://bugs.launchpad.net/zope.configuration/+bug/135267, Christian Zagrodnick reported:

> ZCML directives like `zcml:condition` form the http://namespaces.zope.org/zcml namespace are not in the apidoc's ZCML reference.

@philikon replied:

> `zcml:condition` is not a directive, it's an inherent feature of the configuration machinery.

@freddrake followed up:

> It still needs to be documented in an accessible location.

@miohtama replied:

> The only reference documentation for `zcml:condition` is this bug report.
> 
> I'd like to really find _any_ documentation/source for `zcml:condition`.
> 
> ``` xml
>   <adapter  zcml:condition="installed ore.contentmirror"
>    ...>
> ```

@philikon:

> That doesn't mean this is the right place to ask for more documentation. I recommend writing to the mailto:zope@zope.org or zope3-users@zope.org list.
> 
> My book (http://worldcookery.com) mentions it in Chapter 13 and the ZCML reference (Appendix B).

@benji-york:

> Also see http://svn.zope.org/*checkout*/zope.configuration/trunk/src/zope/configuration/tests/test_conditions.py?content-type=text%2Fplain

@miohtama: 

> Even though this might not be a right place to ask help, I consider it's a bug if the API is not well enough documented. It's just to balancing the sentence structure between "this thing is not properly documented" and "how I work with this because I cannot find any documents."
> 
> This page has some data:
> 
> http://wiki.zope.org/zope3/ZcmlConditionFeatures
> 
> But it fails to mention e.g. has condition "installed zope.app.catalog" like used in here:
> 
> https://textindexng.svn.sourceforge.net/svnroot/textindexng/textindexng/trunk/src/textindexng/configure.zcml
> 
> ZcmConditionFeatures page lacks completely examples.
> 
> Zope API documents at http://apidoc.zope.org/++apidoc++/ don't mention how to use zcml: directives. The reason is explained in this document chain, but is is just an obstacle. How it could be possible to generate API documentation for ZCML inhereted features?
> 
> I was going to create some examples in Zope 3 wiki, but I couldn't find any source material, so I started bragging here.

### PR DESCRIPTION
**Pull request by jamadden**
Also some cross-ref cleanups in narr.rst and zopeconfigure.py

Fixes #5

### CODE REVIEW

**Comment by jamadden** [comment]
Coverage is red here because anticipating #30 , coveralls was set to require 100% coverage.

File: docs/narr.rst [hunk 1]
```diff
@@ -571,16 +571,84 @@ There is a ``condition`` attribute in the
...
+   </configure>
+
+Module Availability
+-------------------
+
+The verbs ``installed`` and ``not-installed`` test whether a Python
+module can be imported successfully or not. They can be used to enable
+optional configuration if a package is present, or enable fallback
+functionality if it isn't. (This also demonstrates grouping directives
+under a new ``<configure>`` element, and establishing a logical "or".)
```
Comment (mgedmin) [inline docs/narr.rst]: I don't think this is a logical "or", more of an if-else.

To me "logical or" would be something like "feature devmode or feature this-specific-thing".

Comment (jamadden) [reply docs/narr.rst]: Good point, I'll clarify.

File: docs/narr.rst [hunk 2]
```diff
@@ -571,16 +571,84 @@ There is a ``condition`` attribute in the
...
+four verbs that are supported:
+
+- have
+- not-have
+- installed
+- not-installed
+
+Note that logical "if/else" conditions can be achieved by using both the
+positive verb and its not-prefixed counterpart on sibling elements
+with the same arguments. Logical "and" conditions can be achieved by
```
Comment (mgedmin) [inline docs/narr.rst]: This was the logical-or sentence that confused me first; I started commenting asking for examples until I noticed there were examples a couple of screens down.

If you're planning to rewrite the logical-or mention there, maybe also update it here.

Comment (jamadden) [reply docs/narr.rst]: I did change it to 'if/else' both locations.

**Comment by jamadden** [comment]
Thanks!

File: docs/api/zopeconfigure.rst [hunk 3]
```diff
@@ -2,11 +2,3 @@
...
 
 .. automodule:: zope.configuration.zopeconfigure
-
-.. autointerface:: IZopeConfigure
-   :members:
-   :member-order: bysource
-
-.. autoclass:: ZopeConfigure
-   :members:
-   :member-order: bysource
```
Comment (tseaver) [inline docs/api/zopeconfigure.rst]: Why delete these here?  Seems unrelated to the PR.

File: docs/narr.rst [hunk 4]
```diff
@@ -571,16 +571,84 @@ There is a ``condition`` attribute in the
...
+nest them under a new ``<configure>`` element.
+
+.. seealso:: :meth:`zope.configuration.xmlconfig.ConfigurationHandler.evaluateCondition`
+
+   For documentation and examples of the XML implementation of conditions.
+
+Features
+--------
+
+The verbs ``have`` and ``not-have`` test for the presence or absense
```
Comment (tseaver) [inline docs/narr.rst]: Typo:  `absence`

File: docs/api/zopeconfigure.rst (hunk 3 above)
Comment (jamadden) [reply docs/api/zopeconfigure.rst]: They were already included from the '.. automodule:: ' above and so were getting documented twice.

File: docs/narr.rst (hunk 4 above)
Comment (jamadden) [reply docs/narr.rst]: Thanks. My spell-checker doesn't flag that for some reason...
